File: AIVA/lib/data/mock/mock_data.dart
import '../models/student_model.dart';
import '../models/lecture_model.dart';
import '../models/doubt_model.dart';
import '../models/course_model.dart';

class MockData {
  static const StudentModel currentStudent = StudentModel(
    id: 's1',
    name: 'Arjun Sharma',
    phone: '+91 98765 43210',
    institute: 'IIT Delhi',
    avatarInitials: 'AS',
    streakDays: 14,
    lecturesWatched: 47,
    doubtsAsked: 128,
  );

  static const List<LectureModel> lectures = [
    LectureModel(
      id: 'l1',
      title: 'React Hooks Deep Dive',
      courseName: 'React.js Mastery',
      instructorName: 'Rahul Gupta',
      duration: '42 min',
      thumbnailGradient: 'indigo',
      progress: 0.65,
      doubtsCount: 23,
      level: 'Intermediate',
      views: 1842,
      description:
          'Master all React hooks including useState, useEffect, useCallback, useMemo, and custom hooks with real-world examples.',
      topics: ['useState', 'useEffect', 'useCallback', 'useMemo', 'Custom Hooks'],
    ),
    LectureModel(
      id: 'l2',
      title: 'Node.js Event Loop Explained',
      courseName: 'Node.js Backend',
      instructorName: 'Priya Singh',
      duration: '38 min',
      thumbnailGradient: 'cyan',
      progress: 0.30,
      doubtsCount: 18,
      isNew: true,
      level: 'Advanced',
      views: 934,
      description:
          'Deep dive into the Node.js event loop, call stack, callback queue, and how async operations are handled.',
      topics: ['Event Loop', 'Callback Queue', 'Promise', 'async/await'],
    ),
    LectureModel(
      id: 'l3',
      title: 'Binary Search Tree Operations',
      courseName: 'DSA with Python',
      instructorName: 'Amit Kumar',
      duration: '55 min',
      thumbnailGradient: 'purple',
      progress: 0.0,
      doubtsCount: 31,
      level: 'Beginner',
      views: 2341,
      description:
          'Understand BST insertion, deletion, traversal and balancing with visual animations.',
      topics: ['BST', 'Traversal', 'Inorder', 'Preorder', 'Postorder'],
    ),
    LectureModel(
      id: 'l4',
      title: 'CSS Grid & Flexbox Mastery',
      courseName: 'Frontend Development',
      instructorName: 'Neha Joshi',
      duration: '31 min',
      thumbnailGradient: 'pink',
      progress: 1.0,
      doubtsCount: 12,
      isNew: true,
      level: 'Beginner',
      views: 3120,
      description:
          'Complete guide to CSS Grid and Flexbox layouts with hands-on projects.',
      topics: ['Grid', 'Flexbox', 'Responsive Design', 'CSS Variables'],
    ),
    LectureModel(
      id: 'l5',
      title: 'Python Decorators & Generators',
      courseName: 'Python Advanced',
      instructorName: 'Vikram Mehta',
      duration: '44 min',
      thumbnailGradient: 'orange',
      progress: 0.50,
      doubtsCount: 19,
      level: 'Advanced',
      views: 1567,
      description:
          'Learn Python decorators, generators, and iterators with practical examples.',
      topics: ['Decorators', 'Generators', 'Yield', 'Iterators', 'Closures'],
    ),
  ];

  static final List<DoubtModel> doubts = [
    DoubtModel(
      id: 'd1',
      question: 'What is the difference between useCallback and useMemo?',
      answer:
          '**useCallback** returns a memoized callback function, while **useMemo** returns a memoized value.\n\nuseCallback prevents a function from being recreated on every render. useMemo caches the result of an expensive calculation.\n\n📍 From 42:31 — In this timestamp, Rahul explains this with a live counter example.',
      lectureId: 'l1',
      lectureName: 'React Hooks Deep Dive',
      timestamp: '42:31',
      askedAt: DateTime(2026, 5, 9, 21, 30),
      language: 'EN',
      isAi: false,
    ),
    DoubtModel(
      id: 'd1_ai',
      question: '',
      answer:
          '**useCallback** returns a memoized callback function, while **useMemo** returns a memoized value.\n\nuseCallback prevents a function from being recreated on every render. useMemo caches the result of an expensive calculation.\n\n📍 From 42:31 — In this timestamp, Rahul explains this with a live counter example.',
      lectureId: 'l1',
      lectureName: 'React Hooks Deep Dive',
      timestamp: '42:31',
      askedAt: DateTime(2026, 5, 9, 21, 30),
      language: 'EN',
      isAi: true,
    ),
    DoubtModel(
      id: 'd2',
      question: 'Event loop kaise kaam karta hai Node.js mein?',
      answer:
          '**Event Loop** Node.js ka core mechanism hai jo asynchronous operations handle karta hai.\n\n1. Call Stack — Synchronous code execute hota hai\n2. Web APIs — setTimeout, fetch async operations yahan jaate hain\n3. Callback Queue — Completed callbacks yahan wait karte hain\n4. Event Loop — Jab stack khali ho, callbacks ko push karta hai\n\n📍 From 12:45 — Is timestamp par diagram ke saath samjhaya gaya hai.',
      lectureId: 'l2',
      lectureName: 'Node.js Event Loop',
      timestamp: '12:45',
      askedAt: DateTime(2026, 5, 9, 17, 0),
      language: 'HI',
      isAi: false,
    ),
    DoubtModel(
      id: 'd3',
      question: 'How does BST deletion work when node has two children?',
      answer:
          'When deleting a node with two children from a BST:\n\n1. Find the inorder successor (smallest node in right subtree)\n2. Copy its value to the node being deleted\n3. Delete the inorder successor from the right subtree\n\n📍 From 28:15 — Visual animation of this process is shown here.\n🔗 Also in Lecture 2 · 15:42 — Similar concept covered for AVL trees.',
      lectureId: 'l3',
      lectureName: 'Binary Search Tree Operations',
      timestamp: '28:15',
      askedAt: DateTime(2026, 5, 8, 22, 0),
      language: 'EN',
      isAi: false,
    ),
  ];

  static const List<CourseModel> courses = [
    CourseModel(
      id: 'c1',
      name: 'React.js Mastery',
      instructor: 'Rahul Gupta',
      totalLectures: 24,
      completedLectures: 15,
      category: 'Frontend',
      progress: 0.625,
    ),
    CourseModel(
      id: 'c2',
      name: 'Node.js Backend',
      instructor: 'Priya Singh',
      totalLectures: 18,
      completedLectures: 5,
      category: 'Backend',
      progress: 0.278,
    ),
    CourseModel(
      id: 'c3',
      name: 'DSA with Python',
      instructor: 'Amit Kumar',
      totalLectures: 32,
      completedLectures: 12,
      category: 'DSA',
      progress: 0.375,
    ),
  ];

  static const List<String> filterTopics = [
    'All', 'React', 'Node.js', 'DSA', 'Python', 'System Design', 'Databases',
  ];

  static const List<Map<String, dynamic>> weeklyActivity = [
    {'day': 'Mon', 'doubts': 5},
    {'day': 'Tue', 'doubts': 8},
    {'day': 'Wed', 'doubts': 3},
    {'day': 'Thu', 'doubts': 12},
    {'day': 'Fri', 'doubts': 7},
    {'day': 'Sat', 'doubts': 15},
    {'day': 'Sun', 'doubts': 9},
  ];

  static const List<String> masteredConcepts = [
    'useState', 'useEffect', 'Props', 'JSX', 'Components',
    'Flexbox', 'Grid', 'CSS Variables', 'Responsive',
    'Arrays', 'Linked Lists', 'Recursion', 'Sorting',
    'Closures', 'Promises', 'async/await',
  ];

  static const List<String> studyingTopics = [
    'React', 'Node.js', 'DSA', 'Python', 'System Design',
  ];
}

File: admin/lib/mock-data.ts
export const mockStats = {
  totalStudents: 1284,
  activeToday: 347,
  doubtsToday: 892,
  lecturesUploaded: 64,
  totalStudentsChange: 12,
  activeTodayChange: -3,
  doubtsTodayChange: 24,
  lecturesUploadedChange: 8,
};

export const mockDoubtsOverTime = [
  { day: "Mon", doubts: 342 },
  { day: "Tue", doubts: 518 },
  { day: "Wed", doubts: 421 },
  { day: "Thu", doubts: 687 },
  { day: "Fri", doubts: 593 },
  { day: "Sat", doubts: 892 },
  { day: "Sun", doubts: 731 },
];

export const mockConfusedTopics = [
  { topic: "useCallback vs useMemo", count: 234 },
  { topic: "Event Loop", count: 189 },
  { topic: "BST Deletion", count: 167 },
  { topic: "Async/Await", count: 143 },
  { topic: "CSS Grid", count: 121 },
];

export const mockRecentDoubts = [
  {
    id: "d1",
    student: "Arjun Sharma",
    lecture: "React Hooks Deep Dive",
    doubt: "What is the difference between useCallback and useMemo?",
    time: "2 min ago",
    status: "answered",
  },
  {
    id: "d2",
    student: "Priya Patel",
    lecture: "Node.js Event Loop",
    doubt: "Event loop kaise kaam karta hai?",
    time: "5 min ago",
    status: "answered",
  },
  {
    id: "d3",
    student: "Rahul Verma",
    lecture: "BST Operations",
    doubt: "How does BST deletion work with two children?",
    time: "12 min ago",
    status: "answered",
  },
  {
    id: "d4",
    student: "Sneha Kumar",
    lecture: "CSS Grid & Flexbox",
    doubt: "What is the difference between grid and flex?",
    time: "18 min ago",
    status: "answered",
  },
  {
    id: "d5",
    student: "Amit Singh",
    lecture: "Python Decorators",
    doubt: "How do closures work in Python decorators?",
    time: "25 min ago",
    status: "pending",
  },
];

export const mockAtRiskStudents = [
  { id: "s1", name: "Vikas Gupta", lastActive: "4 days ago", behind: 3, avatar: "VG" },
  { id: "s2", name: "Meena Joshi", lastActive: "5 days ago", behind: 5, avatar: "MJ" },
  { id: "s3", name: "Rohit Das", lastActive: "7 days ago", behind: 8, avatar: "RD" },
];

export const mockCourses = [
  {
    id: "c1",
    name: "React.js Mastery",
    instructor: "Rahul Gupta",
    lectures: 24,
    students: 487,
    status: "active",
    category: "Frontend",
  },
  {
    id: "c2",
    name: "Node.js Backend",
    instructor: "Priya Singh",
    lectures: 18,
    students: 312,
    status: "active",
    category: "Backend",
  },
  {
    id: "c3",
    name: "DSA with Python",
    instructor: "Amit Kumar",
    lectures: 32,
    students: 623,
    status: "active",
    category: "DSA",
  },
  {
    id: "c4",
    name: "Frontend Development",
    instructor: "Neha Joshi",
    lectures: 20,
    students: 289,
    status: "draft",
    category: "Frontend",
  },
  {
    id: "c5",
    name: "System Design",
    instructor: "Vikram Mehta",
    lectures: 15,
    students: 198,
    status: "active",
    category: "System Design",
  },
  {
    id: "c6",
    name: "Python Advanced",
    instructor: "Ananya Roy",
    lectures: 22,
    students: 0,
    status: "draft",
    category: "Python",
  },
];

export const mockLectures = [
  {
    id: "l1",
    title: "React Hooks Deep Dive",
    course: "React.js Mastery",
    duration: "42:31",
    uploaded: "May 5, 2026",
    doubts: 234,
    views: 1842,
  },
  {
    id: "l2",
    title: "Node.js Event Loop Explained",
    course: "Node.js Backend",
    duration: "38:15",
    uploaded: "May 6, 2026",
    doubts: 189,
    views: 934,
  },
  {
    id: "l3",
    title: "Binary Search Tree Operations",
    course: "DSA with Python",
    duration: "55:02",
    uploaded: "May 7, 2026",
    doubts: 167,
    views: 2341,
  },
  {
    id: "l4",
    title: "CSS Grid & Flexbox Mastery",
    course: "Frontend Development",
    duration: "31:44",
    uploaded: "May 8, 2026",
    doubts: 121,
    views: 3120,
  },
  {
    id: "l5",
    title: "Python Decorators & Generators",
    course: "Python Advanced",
    duration: "44:18",
    uploaded: "May 9, 2026",
    doubts: 143,
    views: 1567,
  },
];

export const mockStudents = [
  {
    id: "s1",
    name: "Arjun Sharma",
    avatar: "AS",
    phone: "+91 98765 43210",
    course: "React.js Mastery",
    progress: 62,
    doubts: 128,
    lastActive: "Just now",
    status: "active",
  },
  {
    id: "s2",
    name: "Priya Patel",
    avatar: "PP",
    phone: "+91 87654 32109",
    course: "Node.js Backend",
    progress: 28,
    doubts: 47,
    lastActive: "1h ago",
    status: "active",
  },
  {
    id: "s3",
    name: "Rahul Verma",
    avatar: "RV",
    phone: "+91 76543 21098",
    course: "DSA with Python",
    progress: 75,
    doubts: 203,
    lastActive: "3h ago",
    status: "active",
  },
  {
    id: "s4",
    name: "Sneha Kumar",
    avatar: "SK",
    phone: "+91 65432 10987",
    course: "Frontend Development",
    progress: 91,
    doubts: 56,
    lastActive: "1d ago",
    status: "inactive",
  },
  {
    id: "s5",
    name: "Vikas Gupta",
    avatar: "VG",
    phone: "+91 54321 09876",
    course: "React.js Mastery",
    progress: 12,
    doubts: 8,
    lastActive: "4d ago",
    status: "at-risk",
  },
  {
    id: "s6",
    name: "Meena Joshi",
    avatar: "MJ",
    phone: "+91 43210 98765",
    course: "DSA with Python",
    progress: 5,
    doubts: 3,
    lastActive: "5d ago",
    status: "at-risk",
  },
];

export const mockDoubtAnalytics = {
  totalDoubts: 5847,
  avgResponseTime: "2.3s",
  hindiPercent: 38,
  resolvedPercent: 97,
};

export const mockConfusionHeatmap = [
  {
    lecture: "React Hooks Deep Dive",
    segments: [2, 5, 3, 8, 12, 18, 9, 15, 7, 4],
  },
  {
    lecture: "Node.js Event Loop",
    segments: [1, 3, 14, 21, 8, 6, 3, 2, 1, 1],
  },
  {
    lecture: "BST Operations",
    segments: [4, 6, 8, 11, 16, 9, 7, 5, 3, 2],
  },
  {
    lecture: "CSS Grid & Flexbox",
    segments: [2, 4, 3, 5, 7, 4, 3, 2, 1, 1],
  },
];

export const mockTopQuestions = [
  { question: "What is the difference between useCallback and useMemo?", count: 234, lecture: "React Hooks", topic: "React" },
  { question: "Event loop kaise kaam karta hai?", count: 189, lecture: "Node.js Event Loop", topic: "Node.js" },
  { question: "How does BST deletion work with two children?", count: 167, lecture: "BST Operations", topic: "DSA" },
  { question: "What is the difference between grid and flex?", count: 143, lecture: "CSS Grid", topic: "CSS" },
  { question: "How do closures work in Python decorators?", count: 121, lecture: "Python Decorators", topic: "Python" },
  { question: "What is async/await and how is it different from Promises?", count: 118, lecture: "Node.js Event Loop", topic: "JavaScript" },
  { question: "Inorder traversal kya hota hai?", count: 97, lecture: "BST Operations", topic: "DSA" },
  { question: "How does useEffect dependency array work?", count: 89, lecture: "React Hooks", topic: "React" },
  { question: "What is memoization?", count: 76, lecture: "React Hooks", topic: "React" },
  { question: "How do Python generators differ from normal functions?", count: 68, lecture: "Python Decorators", topic: "Python" },
];

export const mockDoubtsByTopic = [
  { topic: "React", value: 1842, color: "#6C63FF" },
  { topic: "Node.js", value: 934, color: "#00D4FF" },
  { topic: "DSA", value: 1234, color: "#8B5CF6" },
  { topic: "Python", value: 876, color: "#10B981" },
  { topic: "CSS", value: 543, color: "#F59E0B" },
  { topic: "System Design", value: 418, color: "#EF4444" },
];

export const mockDoubtsByHour = [
  { hour: "6AM", doubts: 12 },
  { hour: "8AM", doubts: 34 },
  { hour: "10AM", doubts: 87 },
  { hour: "12PM", doubts: 124 },
  { hour: "2PM", doubts: 98 },
  { hour: "4PM", doubts: 143 },
  { hour: "6PM", doubts: 201 },
  { hour: "8PM", doubts: 287 },
  { hour: "10PM", doubts: 342 },
  { hour: "12AM", doubts: 198 },
  { hour: "2AM", doubts: 67 },
  { hour: "4AM", doubts: 18 },
];
